server/utils/socketsServer/socketServer.js

```diff
@@ -0,0 +1,91 @@
+const socketio = require("socket.io");
+
+const connectedUsers = {};
+let socket = null;
+let io = null;
+
+const initSocketServer = (server, req) => {
+
+  if(!io) {
+    io = socketio(server, {
+      cors: {
+        origin: "*"
+      }
+    });
+    req.io = io;
+  }
+
+  io.on("connection", _socket => {
+    socket = _socket;
```

Comment:
After removing the global scope, this second variable won't be necessary anymore